### DISCUSSION TITLE
Installtion and run the sourcespec program

### DISCUSSION THREAD

**Discussion by munguiko1** [SeismicSource/sourcespec#60] [Q&A]
Hello, I started the installation and everything is ok until "pip install -e.", the rest I did not succeed, so can someone help me until the program runs?

**Comment by claudiodsf** [comment]
Hello. 

Could you try downloading or cloning the repository again?

There was a problem this morning with the GitHub repository that I fixed a couple of hours ago.

**Reply by munguiko1** [reply]
Thank you, I have recovered the repository files. Unfortunately I have difficulties running the program?

**Reply by claudiodsf** [reply]
If you elaborate more about the problem, maybe we can help you. 

Do you get an error? Do you get a strange result?

**Reply by munguiko1** [reply]
Hello, my sourcespec program is running fine now after many tries, I had to close the line "from scipy.integrate import cumtrapz", that's why it ran. The error can not import "scipy.integrate import cumtrapz" was returned every time I ran the program.
the test-cdsa ab gives a result in sspec_out.
The next step is to know how to use my data. see my output
![cdsa20100421051050GL boxplot](https://github.com/user-attachments/assets/13fdb889-e99b-48da-a07c-2a8758ae5c67)
![cdsa20100421051050GL map_mag](https://github.com/user-attachments/assets/74338d09-2016-4977-be17-efba3cb0457e)

**Reply by claudiodsf** [reply]
Hello @munguiko1,

thanks for reporting: this is a known error that will be fixed in SourceSpec 1.9 which hopefully will be released this week or next week.

Your fix is fine if you don't use time-domain integration. A better fix is to add these two lines in `ssp_build_spectra.py`:

```python
try:
    from scipy.integrate import cumtrapz
except ImportError:
    from scipy.integrate import cumulative_trapezoid as cumtrapz
```

**Reply by claudiodsf** [reply]
Now, coming to your question:

> The next step is to know how to use my data

Which formats do you use for traces, for station response and for event information?

**Comment by munguiko1** [comment]
The traces are in mseed formats, station response in seiscomp

Le lun. 4 nov. 2024 à 09:49, Claudio Satriano ***@***.***> a
écrit :

**Reply by claudiodsf** [reply]
I never tried SeisComP station response format, but it is supported by ObsPy, so it should work.

You can try something like:

```
source_spec -t TRACE_DIR -w RESPONSE_DIR
```

where `TRACE_DIR` and `RESPONSE_DIR` need to be replaced with actual paths to directories where you store traces and station responses.

However, there is still one piece missing: event information. If you don't have any file with event information (e.g., QuakeML), you can build your own using the SourceSpec Event File format: https://sourcespec.readthedocs.io/en/stable/source_spec_event_file.html

**Comment by munguiko1** [comment]
Thank you very much Satriano for your interventions. I am here in
Clermont-Ferrand from the Democratic Republic of Congo, I work on the
Nyiragongo and Nyamulagira Volcanoes. I am not a seismologist, but I have
been very interested in seismology since I started working at the Goma
Volcano Observatory in Seismology Department. Certainly, I have mastered
many concepts but not in the context of more in-depth analyses like those
you do. Here at the Laboratoire Magmas and Volcans (LMV) I am leading a
project on the* "Spatio-temporel variability of diffuse degassing: link
with volcano-tectonic activity"*. This project aims to explore the link
between the degassing of CO2 and radon with the activity of the Nyiragongo
and Nyamulagira volcanoes, and also with tectonic activity, because we are
in an extensional rift. For seismic analysis, I need the analysis to detect
any change in volcanic activity, and here I think about the temporel
evolution of dominant resonance frequencies and amplitudes (especiall of
the volcaniques earthquakes). The calculation of magnitude of regional
earthquakes and to that I may wish to add source parameters (stress drop,
source radius, corner frequency). That's why I put my interest in the
sourcespec program. The project is successful when it has been, not only
shared but received advice, rich remarks and new directions. So, you will
be tolerant to my questions because I want to learn more.
I would like to share and work with you on our tracks (seismic data) which
contain a lot of information, unfortunately not exploited.
Olivier Munguiko

Le lun. 4 nov. 2024 à 12:17, Claudio Satriano ***@***.***> a
écrit :

**Reply by claudiodsf** [reply]
Dear @munguiko1,

Your project is indeed very interesting and would certainly benefit from the expertise of a seismologist.

Using SourceSpec does require a certain level of seismological knowledge. Unfortunately, I am unable to provide the necessary collaboration at this time.

The objective of this forum is to offer technical support and facilitate brief discussions on scientific matters. Establishing a scientific collaboration goes beyond the forum's intended scope.

However, you are welcome to continue seeking technical support here.

Claudio
